eng/common/scripts/Service-Level-Readme-Automation.ps1

```diff
@@ -0,0 +1,233 @@
+<#
+.SYNOPSIS
+The script is to generate service level readme if it is missing. 
+For exist ones, we do 2 things here:
+1. Generate the client but not import to the existing service level readme.
+2. Update the metadata of service level readme
+
+.DESCRIPTION
+Given a doc repo location, and the credential for fetching the ms.author. 
+Generate missing service level readme and updating metadata of the existing ones.
+
+.PARAMETER DocRepoLocation
+Location of the documentation repo. This repo may be sparsely checked out
+depending on the requirements for the domain
+
+.PARAMETER TenantId
+The aad tenant id/object id for ms.author.
+
+.PARAMETER ClientId
+The add client id/application id for ms.author.
+
+.PARAMETER ClientSecret
+The client secret of add app for ms.author.
+#>
+
+param(
+  [Parameter(Mandatory = $true)]
+  [string] $DocRepoLocation,
+
+  [Parameter(Mandatory = $false)]
+  [string]$TenantId,
+
+  [Parameter(Mandatory = $false)]
+  [string]$ClientId,
+
+  [Parameter(Mandatory = $false)]
+  [string]$ClientSecret
+)
+. $PSScriptRoot/common.ps1
+. $PSScriptRoot/Helpers/Metadata-Helpers.ps1
+. $PSScriptRoot/Helpers/Package-Helpers.ps1
+
+Set-StrictMode -Version 3
+
+function create-metadata-table($readmeFolder, $readmeName, $moniker, $msService, $clientTableLink, $mgmtTableLink, $serviceName)
+{
+  $readmePath = Join-Path $readmeFolder -ChildPath $readmeName
+  $content = ""  
+  if (Test-Path (Join-Path $readmeFolder -ChildPath $clientTableLink)) {
+    $content = "## Client packages - $moniker`r`n"
+    $content += "[!INCLUDE [client-packages]($clientTableLink)]`r`n"
+  }
+  if (Test-Path (Join-Path $readmeFolder -ChildPath $mgmtTableLink)) {
+    $content = "## Management packages - $moniker`r`n"
+    $content += "[!INCLUDE [mgmt-packages]($mgmtTableLink)]`r`n"
+  }
+  if (!$content) {
+    return
+  }
+  $null = New-Item -Path $readmePath -Force
+  $lang = $LanguageDisplayName
+  $langTitle = "Azure $serviceName SDK for $lang"
+  $header = GenerateDocsMsMetadata -language $lang -langTitle $langTitle -serviceName $serviceName `
```

Comment:
Just FYI. this "header" thing is called the yml `front-matter` by the docs team. No reason to rename at this point, but I'll remind on your next pass over this code.

Reply:
Rename is not a big deal. Will align with the docs and add comments for reference.